### PR TITLE
feat: add plugin kubectl-blame

### PR DESCRIPTION
**Pull request by knight42**
This plugin enable users to visualize the manager of each field of the given resource.

### CODE REVIEW

**Comment by ahmetb** [comment]
Last time we discussed this https://github.com/kubernetes-sigs/krew-index/pull/1008#issuecomment-777671858 it was someone else who submitted your plugin on your behalf, that's why it was rejected. Reviewing now.

File: plugins/blame.yaml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+apiVersion: krew.googlecontainertools.github.com/v1alpha2
+kind: Plugin
+metadata:
+  name: blame
+spec:
+  version: v0.0.5
+  homepage: https://github.com/knight42/kubectl-blame
+  shortDescription: Show the manager of each field of given resource with information from managed fields.
```
Comment (ahmetb) [inline plugins/blame.yaml:8]: > Show the manager of each field of given resource with information from managed fields.

unfortunately this is too long to fit. Limit to 50 chars if you can. e.g. `Show who edited resource fields` is better.

> with information from managed fields.

"How" you do it is a detail, so it should go in `description`.

File: plugins/blame.yaml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  description: |
+    Annotate each line in the given resource's YAML with information from the managedFields to show who last modified the field.
+
+    Examples:
+
+    # Blame pod 'foo' in default namespace
+    kubectl blame pods foo
+
+    # Blame deployment 'bar' in 'ns1' namespace
+    kubectl blame -n ns1 deploy bar
```
Comment (ahmetb) [inline plugins/blame.yaml:18]: typically it's better if you repo or --help message contains these examples.
You can remove these.

File: plugins/blame.yaml [hunk 3]
```diff
@@ -0,0 +1,33 @@
+apiVersion: krew.googlecontainertools.github.com/v1alpha2
+kind: Plugin
+metadata:
+  name: blame
+spec:
+  version: v0.0.5
+  homepage: https://github.com/knight42/kubectl-blame
+  shortDescription: Show the manager of each field of given resource with information from managed fields.
+  description: |
+    Annotate each line in the given resource's YAML with information from the managedFields to show who last modified the field.
```
Comment (ahmetb) [inline plugins/blame.yaml:10]: Please line wrap at 80 chars.

**Comment by ahmetb** [comment]
Mostly good. Only cosmetic issues.
/approve

**Comment by knight42** [comment]
Hi @ahmetb! Thanks for your review, please take another look!

**Comment by knight42** [comment]
> please consider setting up Krew release automation

@ahmetb I am really grateful for letting me know such useful bot! I will set it up later.

> but I saw kubectl commands and other component names

Yeah that is what is saved in the managedFields, for instance:
```
apiVersion: apps/v1
kind: Deployment
metadata:
  labels:
    app: iperf
  managedFields:
  - apiVersion: apps/v1
    fieldsType: FieldsV1
    fieldsV1:
      f:spec:
        f:replicas: {}
    manager: kubectl-edit
    operation: Update
    time: "2021-08-12T16:38:32Z"
```

The owner of the field is actually the manager saved in managedFields. The kubectl commands means the fields were updated by the corresponding command and component names means the fields were updated by the corresponding component 😉 .

**Pull request merged**